skimage/filters/thresholding.py

```diff
@@ -134,6 +135,59 @@ def wrapper(im):
                     methods=methods, verbose=verbose)
 
 
+def try_all_threshold_dict(image):
+    """Returns a dictionary comparing the output of different threshold methods.
+
+    Parameters
+    ----------
+    image : (N, M) ndarray
+        Input image.
+
+    Returns
+    -------
+    threshold_results : dict
+        Dictionary of threshold results in format: {'threshold_method': result}
+        If an exception is encountered in a thresholding method, None is given
+        for that method.
+
+    Notes
+    -----
+    The following algorithms are used:
+
+    * isodata
+    * li
+    * mean
+    * minimum
+    * otsu
+    * triangle
+    * yen
+
+    Examples
+    --------
+    >>> from skimage.data import text
+    >>> dict = try_all_threshold_dict(text())
+    """
+    # Global algorithms.
+    methods = OrderedDict({'Isodata': threshold_isodata,
```

Comment:
It would make sense to me to do not duplicate this list in the code in order to make the addition of algorithms more straightfoward.